Bottom: 579709ad45c967007a3264fcdf2cd4198f841fa2
Top:    996db1b717e67a922e484dda5fc9f6d62c65d5c8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 18:44:41 +0000

find.c: Remove "flag" variable in DRCAll(), and pass explicitly each time.

This avoids confusion by making the flag passed to each call clearer.
It removes several cases where we would set the variable, use it once,
then change it for the next usage, making the code more succinct.


---

diff --git a/src/find.c b/src/find.c
index 184e001..05081b0 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3577,7 +3577,6 @@ DRCAll (void)
   int tmpcnt;
   int nopastecnt = 0;
   bool IsBad;
-  int flag;
   struct drc_info info;
 
   reset_drc_dialog_message();
@@ -3589,9 +3588,7 @@ DRCAll (void)
   hid_action ("LayersChanged");
   InitConnectionLookup ();
 
-  flag = FOUNDFLAG | DRCFLAG | SELECTEDFLAG;
-
-  if (ClearFlagOnAllObjects (true, flag))
+  if (ClearFlagOnAllObjects (true, FOUNDFLAG | DRCFLAG | SELECTEDFLAG))
     {
       IncrementUndoSerialNumber ();
       Draw ();
@@ -3644,10 +3641,8 @@ DRCAll (void)
   }
   END_LOOP;
 
-  flag = (IsBad) ? DRCFLAG : (FOUNDFLAG | DRCFLAG | SELECTEDFLAG);
-  ClearFlagOnAllObjects (false, flag);
-  flag = SELECTEDFLAG;
-  info.flag = flag;
+  ClearFlagOnAllObjects (false, IsBad ? DRCFLAG : (FOUNDFLAG | DRCFLAG | SELECTEDFLAG));
+  info.flag = SELECTEDFLAG;
   /* check minimum widths and polygon clearances */
   if (!IsBad)
     {
@@ -3662,7 +3657,7 @@ DRCAll (void)
         if (line->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (LINE_TYPE, layer, line, line);
-            SET_FLAG (flag, line);
+            SET_FLAG (SELECTEDFLAG, line);
             DrawLine (layer, line);
             drcerr_count++;
             SetThing (LINE_TYPE, layer, line, line);
@@ -3706,7 +3701,7 @@ DRCAll (void)
         if (arc->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (ARC_TYPE, layer, arc, arc);
-            SET_FLAG (flag, arc);
+            SET_FLAG (SELECTEDFLAG, arc);
             DrawArc (layer, arc);
             drcerr_count++;
             SetThing (ARC_TYPE, layer, arc, arc);
@@ -3751,7 +3746,7 @@ DRCAll (void)
             pin->Thickness - pin->DrillingHole < 2 * PCB->minRing)
           {
             AddObjectToFlagUndoList (PIN_TYPE, element, pin, pin);
-            SET_FLAG (flag, pin);
+            SET_FLAG (SELECTEDFLAG, pin);
             DrawPin (pin);
             drcerr_count++;
             SetThing (PIN_TYPE, element, pin, pin);
@@ -3783,7 +3778,7 @@ DRCAll (void)
         if (pin->DrillingHole < PCB->minDrill)
           {
             AddObjectToFlagUndoList (PIN_TYPE, element, pin, pin);
-            SET_FLAG (flag, pin);
+            SET_FLAG (SELECTEDFLAG, pin);
             DrawPin (pin);
             drcerr_count++;
             SetThing (PIN_TYPE, element, pin, pin);
@@ -3826,7 +3821,7 @@ DRCAll (void)
         if (pad->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (PAD_TYPE, element, pad, pad);
-            SET_FLAG (flag, pad);
+            SET_FLAG (SELECTEDFLAG, pad);
             DrawPad (pad);
             drcerr_count++;
             SetThing (PAD_TYPE, element, pad, pad);
@@ -3871,7 +3866,7 @@ DRCAll (void)
             via->Thickness - via->DrillingHole < 2 * PCB->minRing)
           {
             AddObjectToFlagUndoList (VIA_TYPE, via, via, via);
-            SET_FLAG (flag, via);
+            SET_FLAG (SELECTEDFLAG, via);
             DrawVia (via);
             drcerr_count++;
             SetThing (VIA_TYPE, via, via, via);
@@ -3903,7 +3898,7 @@ DRCAll (void)
         if (via->DrillingHole < PCB->minDrill)
           {
             AddObjectToFlagUndoList (VIA_TYPE, via, via, via);
-            SET_FLAG (flag, via);
+            SET_FLAG (SELECTEDFLAG, via);
             DrawVia (via);
             drcerr_count++;
             SetThing (VIA_TYPE, via, via, via);
@@ -3936,19 +3931,17 @@ DRCAll (void)
     }
 
   FreeConnectionLookupMemory ();
-  flag = FOUNDFLAG;
   Bloat = 0;
 
   /* check silkscreen minimum widths outside of elements */
   /* XXX - need to check text and polygons too! */
-  flag = SELECTEDFLAG;
   if (!IsBad)
     {
       SILKLINE_LOOP (PCB->Data);
       {
         if (line->Thickness < PCB->minSlk)
           {
-            SET_FLAG (flag, line);
+            SET_FLAG (SELECTEDFLAG, line);
             DrawLine (layer, line);
             drcerr_count++;
             SetThing (LINE_TYPE, layer, line, line);
@@ -3981,7 +3974,6 @@ DRCAll (void)
 
   /* check silkscreen minimum widths inside of elements */
   /* XXX - need to check text and polygons too! */
-  flag = SELECTEDFLAG;
   if (!IsBad)
     {
       ELEMENT_LOOP (PCB->Data);
@@ -4000,7 +3992,7 @@ DRCAll (void)
             char *buffer;
             int buflen;
 
-            SET_FLAG (flag, element);
+            SET_FLAG (SELECTEDFLAG, element);
             DrawElement (element);
             drcerr_count++;
             SetThing (ELEMENT_TYPE, element, element, element);
